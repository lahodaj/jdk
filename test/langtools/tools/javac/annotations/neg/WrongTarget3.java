/*
 * @test /nodynamiccopyright/
 * @compile/fail/ref=WrongTarget3.out -XDrawDiagnostics WrongTarget3.java
 * @enablePreview
 */

import static java.lang.annotation.ElementType.DECONSTRUCTOR;
import static java.lang.annotation.ElementType.METHOD;

@java.lang.annotation.Target({METHOD})
@interface Wrong {
}

@java.lang.annotation.Target({DECONSTRUCTOR})
@interface OK {
}

public class WrongTarget3 {

    @Wrong
    public pattern WrongTarget3(String name) {
        match WrongTarget3("");
    }

    @OK
    public pattern WrongTarget3(String name, int i) {
        match WrongTarget3("", 42);
    }
}
