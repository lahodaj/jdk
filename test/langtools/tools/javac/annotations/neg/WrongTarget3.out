WrongTarget3.java:20:5: compiler.err.annotation.type.not.applicable
- compiler.note.preview.filename: WrongTarget3.java, DEFAULT
- compiler.note.preview.recompile
1 error